include/libgit4cpp/Remote.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * \file   Remote.h
```

Comment:
I find `git::GitRepository` redundant and I would actually prefer to rename `git::GitRepository->git::Repository` instead of `git::Remote->git::GitRemote`.

Reply:
Then lets do that (in a next PR)